### ISSUE TITLE
Encode Decimal into query parameters

### ISSUE THREAD

**Issue by pykaso** [Alamofire/Alamofire#3184]
## What did you do?

Encode structure bellow in query parameters with `URLEncodedFormEncoder`.

```
   struct SampleStruct: Encodable {
      let price: Decimal
      let discount: Decimal
   }

   let data = SampleStruct(price: 12.34, discount: 0.01)
```

## What did you expect to happen?

Url with query part `?price=12.34&discount=0.01`.

## What happened instead?

Encoded result:

```
discount%5B%5D%5B%5D=0&discount%5B%5D%5B%5D=0&discount%5B%5D%5B%5D=0&discount%5B%5D%5B%5D=0&discount%5B%5D%5B%5D=0&discount%5B%5D%5B%5D=0&discount%5B%5D%5B%5D=0&discount%5B%5D%5B%5D=1&discount%5B%5D=-2&discount%5B%5D=0&discount%5B%5D=1&discount%5B%5D=1&price%5B%5D%5B%5D=0&price%5B%5D%5B%5D=0&price%5B%5D%5B%5D=0&price%5B%5D%5B%5D=0&price%5B%5D%5B%5D=0&price%5B%5D%5B%5D=0&price%5B%5D%5B%5D=0&price%5B%5D%5B%5D=1234&price%5B%5D=-2&price%5B%5D=0&price%5B%5D=1&price%5B%5D=1

```

## Alamofire Environment

**Alamofire version:** 5.1
**Xcode version:** 11.4
**Swift version:** 5.2
**Platform(s) running Alamofire:** iOS
**macOS version running Xcode:** Catalina

## Demo Project

https://github.com/pykaso/decimal_alamofire_issue

**Comment by jshier** [comment]
Confirmed. Thanks for the report, I'll look into it. The output is rather bizarre, but probably due to not explicitly handling `Decimal` in the encoder.

**Comment by jshier** [comment]
I've put up a PR to fix this. However, you need to be careful when encoding and parsing `Decimal` values, as they do *not* losslessly encode and decode from string values, unfortunately. Converting to and from `Double` when encoding and decoding may actually give you better results.

**Comment by jshier** [comment]
This has been fixed in Alamofire 5.2.
